### PR TITLE
Use CUE to define schema for files

### LINKED ISSUE #345: Introduce new package spec version
Introduce package spec format version example `format_version: '1.0'`

Per discussed during one of our team weekly, we will upgrade the package spec version as It is almost impossible to add control over package spec.

Most of the time it breaks a lot of packages and will imply bulk changes a the same time.

Changes packages spec will help us be more strict on validation for new incoming packages, leave older packages as is, add new types and fields without impacting packages without format_version.

Related:
https://github.com/elastic/package-spec/issues/313
https://github.com/elastic/package-spec/pull/338
https://github.com/elastic/package-spec/pull/281

### PR DESCRIPTION
**Pull request by jsoriano**
## What does this PR do?

Uses CUE to define the schema for files instead of JSON schema (only integration manifest and changelog by now).

It includes https://github.com/elastic/package-spec/pull/384 for semantic rules versioning.

## Thoughts on using CUE for schema validation

TLDR; I am not convinced that it provides general advantages.

Pros:
* Expressive and compact syntax.
* Conditional code.
* Nice features like checking backwards compatibility.

Cons:
* No first-level support for descriptions, examples and so on, we would need to work on something else for documentation.
* Only approach for reusability of definitions is the use of packages, but packages loading don't support `fs.FS` (https://github.com/cue-lang/cue/issues/607). Hacks needed.
* Syntax behaves a bit unexpectedly sometimes. Examples:
  * Data injection with tags is quite tricky, and doesn't always work.
  * Additional contents are only checked for named definitions, they are not checked when validating against the spec in the root of a file. This forces us to work with definitions, not such a big issue.
* Internals are hidden in `internal` libraries. No custom Go logic or types can be added, we'd need to fork or continue with semantic validation. Same limitations in that sense than jsonschema at the end.
* Validation of relative paths and data stream names would need to be done with semantic validation (current approach of using custom "formats" in JSON schema is tricky in any case).
* Errors are sometimes quite cryptic, at least in the context of schema validation. Transformations needed.
* Not 1.0 yet, Go APIs may change.

## Why is it important?

To support multiple versions of the spec with the same files.

## Checklist

- [ ] I have added test packages to [`test/packages`](https://github.com/elastic/package-spec/tree/main/test/packages) that prove my change is effective.
- [ ] I have added an entry in [`versions/N/changelog.yml`](https://github.com/elastic/package-spec/blob/main/versions/1/changelog.yml).
- [x] Validate data stream names.
- [x] Validate relative paths.

## Related issues

- Alternative to #380.
- Includes #384.
- Related to https://github.com/elastic/package-spec/issues/287.
- Closes #345.

### CODE REVIEW

**Comment by jsoriano** [comment]
Closing in favor of https://github.com/elastic/package-spec/pull/384.
